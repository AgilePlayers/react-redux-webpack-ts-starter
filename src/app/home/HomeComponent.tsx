import React from 'react';
import CounterContainer from './components/counter/CounterContainer';

const HomeComponent: React.SFC = () => (
  <>
    <h1>Counter</h1>
    <CounterContainer />
  </>
);

export default HomeComponent;
