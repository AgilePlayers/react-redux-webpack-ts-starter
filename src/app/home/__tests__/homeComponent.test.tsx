import React from 'react';
import { shallow, ShallowWrapper } from 'enzyme';
import HomeComponent from '../HomeComponent';

describe('home / HomeComponent', () => {
  let component: ShallowWrapper;

  beforeEach(() => {
    component = shallow(<HomeComponent />);
  });

  test('should render', () => {
    expect(component.length).toEqual(1);
  });

  test('should match snapshot', () => {
    expect(component).toMatchSnapshot();
  });

  test('should contains h1 element', () => {
    expect(component.find('h1').length).toEqual(1);
  });
});
