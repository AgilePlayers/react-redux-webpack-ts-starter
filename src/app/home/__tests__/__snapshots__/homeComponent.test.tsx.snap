// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`home / HomeComponent should match snapshot 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <HomeComponent />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "function",
    "props": Object {
      "children": Array [
        <h1>
          Counter
        </h1>,
        <Connect(CounterComponent) />,
      ],
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "Counter",
        },
        "ref": null,
        "rendered": "Counter",
        "type": "h1",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {},
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
    ],
    "type": Symbol(react.fragment),
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "function",
      "props": Object {
        "children": Array [
          <h1>
            Counter
          </h1>,
          <Connect(CounterComponent) />,
        ],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Counter",
          },
          "ref": null,
          "rendered": "Counter",
          "type": "h1",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {},
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
      ],
      "type": Symbol(react.fragment),
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
<<<<<<< HEAD:src/app/home/__tests__/__snapshots__/homeComponent.test.tsx.snap
          "getSnapshotBeforeUpdate": true,
=======
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
>>>>>>> npm-update:src-ts/app/home/__tests__/__snapshots__/homeComponent.test.tsx.snap
        },
      },
    },
    "attachTo": undefined,
    "hydrateIn": undefined,
  },
}
`;
